app/uk/gov/hmrc/economiccrimelevyregistration/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -16,53 +16,81 @@
+    (CustomHeaderNames.CorrelationId, correlationId),
+    (HEADER_X_CORRELATION_ID, correlationId)
   )
 
   def getSubscriptionStatus(
     businessPartnerId: String
-  )(implicit hc: HeaderCarrier): Future[SubscriptionStatusResponse] =
-    httpClient.GET[SubscriptionStatusResponse](
-      s"${appConfig.integrationFrameworkUrl}/cross-regime/subscription/ECL/SAFE/$businessPartnerId/status",
-      headers = integrationFrameworkHeaders(appConfig.getSubscriptionStatusBearerToken)
-    )
+  )(implicit hc: HeaderCarrier): Future[SubscriptionStatusResponse] = {
+    val correlationId = createCorrelationId(hc)
+
+    retryFor[SubscriptionStatusResponse]("Get subscription status")(retryCondition) {
+      httpClient
+        .get(url"${appConfig.integrationFrameworkUrl}/cross-regime/subscription/ECL/SAFE/$businessPartnerId/status")
+        .setHeader(integrationFrameworkHeaders(correlationId): _*)
+        .executeAndDeserialise[SubscriptionStatusResponse]
+    }
+  }
 
   def subscribeToEcl(eclSubscription: EclSubscription)(implicit
     hc: HeaderCarrier
-  ): Future[Either[UpstreamErrorResponse, CreateEclSubscriptionResponse]] =
-    httpClient.POST[Subscription, Either[UpstreamErrorResponse, CreateEclSubscriptionResponse]](
-      s"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/${eclSubscription.businessPartnerId}",
-      eclSubscription.subscription,
-      headers = integrationFrameworkHeaders(appConfig.integrationFrameworkBearerToken)
-    )
+  ): Future[CreateEclSubscriptionResponse] = {
+    val correlationId: String = createCorrelationId(hc)
+
+    retryFor[CreateEclSubscriptionResponse]("Subscribe to ECL")(retryCondition) {
+      httpClient
+        .post(
+          url"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/${eclSubscription.businessPartnerId}"
+        )
+        .withBody(Json.toJson(eclSubscription))
+        .setHeader(integrationFrameworkHeaders(correlationId): _*)
+        .executeAndDeserialise[CreateEclSubscriptionResponse]
+    }
+  }
+
+  private def createCorrelationId(hc: HeaderCarrier): String =
+    hc.headers(Seq(CorrelationIdHelper.HEADER_X_CORRELATION_ID)) match {
+      case Nil                     => UUID.randomUUID().toString
+      case Seq((_, correlationId)) => correlationId
+    }
 
   def getSubscription(
     eclReference: String
   )(implicit hc: HeaderCarrier): Future[GetSubscriptionResponse] =
-    httpClient.GET[GetSubscriptionResponse](
-      s"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/$eclReference",
-      headers = integrationFrameworkHeaders(appConfig.integrationFrameworkGetSubscriptionBearerToken)
-    )
+    retryFor[GetSubscriptionResponse]("Get subscription")(retryCondition) {
+      httpClient
+        .get(url"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/$eclReference")
+        .setHeader(integrationFrameworkHeaders(appConfig.integrationFrameworkGetSubscriptionBearerToken): _*)
```

Comment:
You are setting the correllationId parameter to the bearer token here. Needs fixing ASAP.

Reply:
Good spot